exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2780 @@
+	}
+	if count != 0 {
+		params.Set("count", strconv.FormatInt(count, 10))
+	}
+	var resp *IndexDeliveryPrice
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL, common.EncodeURLValues(getDeliveryPrices, params), &resp)
+}
+
+// GetFundingChartData gets funding chart data for the requested instrument and time length
+// supported lengths: 8h, 24h, 1m <-(1month)
+func (d *Deribit) GetFundingChartData(ctx context.Context, instrument, length string) (*FundingChartData, error) {
+	params, err := checkInstrument(instrument)
+	if err != nil {
+		return nil, err
+	}
+	params.Set("length", length)
+	var resp *FundingChartData
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getFundingChartData, params), &resp)
+}
+
+// GetFundingRateHistory retrieves hourly historical interest rate for requested PERPETUAL instrument.
+func (d *Deribit) GetFundingRateHistory(ctx context.Context, instrumentName string, startTime, endTime time.Time) ([]FundingRateHistory, error) {
+	params, err := checkInstrument(instrumentName)
+	if err != nil {
+		return nil, err
+	}
+	err = common.StartEndTimeCheck(startTime, endTime)
+	if err != nil {
+		return nil, err
+	}
+	params.Set("start_timestamp", strconv.FormatInt(startTime.UnixMilli(), 10))
+	params.Set("end_timestamp", strconv.FormatInt(endTime.UnixMilli(), 10))
+	var resp []FundingRateHistory
+	return resp, d.SendHTTPRequest(ctx, exchange.RestSpot, nonMatchingEPL, common.EncodeURLValues(getFundingRateHistory, params), &resp)
+}
+
+// GetFundingRateValue gets funding rate value data.
+func (d *Deribit) GetFundingRateValue(ctx context.Context, instrument string, startTime, endTime time.Time) (float64, error) {
+	params, err := checkInstrument(instrument)
+	if err != nil {
+		return 0, err
+	}
+	err = common.StartEndTimeCheck(startTime, endTime)
+	if err != nil {
+		return 0, err
+	}
+	params.Set("start_timestamp", strconv.FormatInt(startTime.UnixMilli(), 10))
+	params.Set("end_timestamp", strconv.FormatInt(endTime.UnixMilli(), 10))
+	var resp float64
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getFundingRateValue, params), &resp)
+}
+
+// GetHistoricalVolatility gets historical volatility data
+func (d *Deribit) GetHistoricalVolatility(ctx context.Context, ccy string) ([]HistoricalVolatilityData, error) {
+	if ccy == "" {
+		return nil, fmt.Errorf("%w '%s'", errInvalidCurrency, ccy)
+	}
+	params := url.Values{}
+	params.Set("currency", ccy)
+	var data [][2]interface{}
+	err := d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getHistoricalVolatility, params), &data)
+	if err != nil {
+		return nil, err
+	}
+	resp := make([]HistoricalVolatilityData, len(data))
+	for x := range data {
+		timeData, ok := data[x][0].(float64)
+		if !ok {
+			return resp, fmt.Errorf("%v GetHistoricalVolatility: %w for time", d.Name, errTypeAssert)
```

Comment:
There is a common package function for type assert errors `GetTypeAssertError()` that can accommodate and standardise this.